ehrql/loaders.py

```diff
@@ -59,6 +88,70 @@ def load_definition_in_subprocess(definition_type, definition_file, user_args, e
     return deserialize(result.stdout)
 
 
+def isolation_is_supported():
+    return subprocess.run([PLEDGE_BIN, "-T", "pledge"]).returncode == 0
+
+
+def subprocess_run_isolated(args, **kwargs):
+    return subprocess.run(
+        [
+            PLEDGE_BIN,
+            # Configure "pledge" to restrict avilable operations to the minimum needed
+            # to execute a definition file and serialize the result, which are:
+            # - stdio: allow stdio and benign system calls
+            # - rpath: read-only path ops
+            # - prot_exec: allow creating executable memory
+            #
+            # In particular, anything related to networking or socket communication,
+            # executing other processes, and any filesystem modifications are all
+            # disallowed. For a full list of the syscalls implied by these settings see:
+            # https://justine.lol/pledge/#promises
+            "-p",
+            "stdio rpath prot_exec",
+            # Disable "unveil": this would give us more fine-grained control over access
+            # to different paths. But it uses a different mechanism from that used by
+            # "pledge" (specifically, the Landlock kernel module) and for whatever
+            # reason it doesn't seem to available to unprivileged docker containers in
+            # our current production VM. Possibly a docker upgrade will fix this; but as
```

Comment:
Is there a breadcrumb trail somewhere from an issue covering that Docker upgrade?

Reply:
Good suggestion, there is now.